applications/nrf5340_audio/src/bluetooth/ble_audio_services.c

```diff
@@ -336,131 +321,197 @@ static void mcs_local_player_instance_cb(struct media_player *player, int err)
 	 */
 	(void)media_proxy_ctrl_send_command(local_player, &cmd);
 }
-#endif /* (CONFIG_BT_MCS) */
 
 int ble_vcs_vol_set(uint8_t volume)
 {
-#if (CONFIG_BT_VCP_VOL_CTLR)
-	int ret;
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR)) {
+		int ret;
 
-	for (int i = 0; i < CONFIG_BT_MAX_CONN; i++) {
-		ret = ble_vcs_client_remote_set(i);
-		/* If remote peer hasn't been connected before, just skip the operation for it */
-		if (ret == -EINVAL) {
-			continue;
+		for (int i = 0; i < CONFIG_BT_MAX_CONN; i++) {
+			ret = ble_vcs_client_remote_set(i);
+			/* If remote peer hasn't been connected before, just skip the operation */
+			if (ret == -EINVAL) {
+				continue;
+			}
+
+			ERR_CHK_MSG(ret, "Failed to set VCS client to remote device properly");
+			ret = bt_vcp_vol_ctlr_set_vol(vcs, volume);
+			if (ret) {
+				LOG_WRN("Failed to set volume for remote channel %d, ret = %d", i,
+					ret);
+			}
 		}
-		ERR_CHK_MSG(ret, "Failed to set VCS client to remote device properly");
-		ret = bt_vcp_vol_ctlr_set_vol(vcs, volume);
-		if (ret) {
-			LOG_WRN("Failed to set volume for remote channel %d, ret = %d", i, ret);
+
+		return 0;
+	} else if (IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) {
+		return bt_vcp_vol_rend_set_vol(volume);
+	}
+
+	if (CONFIG_AUDIO_DEV == GATEWAY) {
+		if (!IS_ENABLED(CONFIG_STREAM_BIDIRECTIONAL) ||
+		    !IS_ENABLED(CONFIG_AUDIO_SOURCE_I2S)) {
+			LOG_DBG("Vol unchanged. VCP control disabled, GW unidir and/or USB source");
+			return 0;
 		}
 	}
-	return 0;
-#elif (CONFIG_BT_VCP_VOL_REND)
-	return bt_vcp_vol_rend_set_vol(volume);
-#endif /* (CONFIG_BT_VCP_VOL_CTLR) */
-	return -ENXIO;
+
+	LOG_DBG("VCP volume controller not enabled, setting volume locally");
```

Comment:
```suggestion
	LOG_DBG("VCP not enabled, setting volume locally");
```

Reply:
Same for the other logs that are the same

Reply:
Changed